integration/query_comparision_test.go

```diff
@@ -39,22 +39,37 @@ func TestQueryComparisonImplicit(t *testing.T) {
 		filter      bson.D
 		expectedIDs []any
 	}{
+		"Document": {
+			filter:      bson.D{{"value", bson.D{{"foo", int32(42)}, {"42", "foo"}, {"array", bson.A{int32(42), "foo", nil}}}}},
+			expectedIDs: []any{"document-composite"},
+		},
+
+		"Array": {
+			filter:      bson.D{{"value", bson.A{int32(42), "foo", nil}}},
+			expectedIDs: []any{"array-three"},
+		},
+
```

Comment:
No empty line there; we group by argument type

Reply:
fixed